src/main/java/org/fountainmc/forge/mixin/entity/MixinEntity.java

```diff
@@ -24,32 +24,159 @@
+        this.setPosition(destination.getX(), destination.getY(),
+                destination.getZ());
+    }
+
+    @Override
+    public float getPitch() {
+        return this.rotationPitch;
+    }
+
+    @Override
+    public void setPitch(float pitch) {
+        this.rotationPitch = pitch;
+    }
+
+    @Override
+    public float getYaw() {
+        return this.rotationYaw;
+    }
+
+    @Override
+    public void setYaw(float yaw) {
+        this.rotationYaw = yaw;
+    }
+
+    @Override
     public boolean isOnGround() {
         return this.onGround;
     }
+
+    @Nullable
+    @Override
+    public Entity getPrimaryPassenger() {
+        if (getPassengers().size() != 0) {
+            return getPassengers().get(0);
+        }
+        return null;
+    }
+
+    @Nonnull
+    public ImmutableList<Entity> entity$getPassengers() {
+        List<net.minecraft.entity.Entity> entities = this.shadow$getPassengers();
+        List<Entity> apiEntities = new ArrayList<>();
+        for (net.minecraft.entity.Entity entity : entities) {
+            apiEntities.add((Entity) entity);
+        }
+        return ImmutableList.copyOf(apiEntities);
+    }
+
+    @Override
+    public void setPrimaryPassenger(Entity passenger) {
+        this.ejectPassenger(this.getPrimaryPassenger());
+        ((net.minecraft.entity.Entity) passenger).startRiding(this.getCommandSenderEntity());
+    }
+
+    @Override
+    public boolean addPassenger(Entity entity, boolean force) {
+        this.shadow$addPassenger((net.minecraft.entity.Entity) entity);
+        return true;
+    }
+
+    @Override
+    public void ejectAll() {
+        this.getPassengers().forEach(this::ejectPassenger);
+    }
+
+    @Override
+    public void ejectPassenger(Entity passenger) {
+        passenger.leaveVehicle();
+    }
+
+    @Override
+    public int getMaximumPassengers() {
+        return this.shadow$getPassengers().size();
+    }
+
+    @Override
+    public void dismountVehicle() {
+        if (this.shadow$getPassengers().size() != 0) { return; }
+        this.getPrimaryPassenger().leaveVehicle();
+    }
+
+    @Nullable
+    @Override
+    public Entity getVehicle() {
+        return (Entity) this.getCommandSenderEntity().getRidingEntity();
+    }
+
+    @Override
+    public void leaveVehicle() {
+        (this.getCommandSenderEntity()).dismountRidingEntity();
+    }
+
+    @Override
+    public ImmutableCollection<Entity> getNearbyEntities(double distance) {
+        List<net.minecraft.entity.Entity> nmsEntities = this.getCommandSenderEntity().worldObj.getEntitiesWithinAABBExcludingEntity(this.getCommandSenderEntity(), this.getCommandSenderEntity().getEntityBoundingBox().expand(distance / 3, distance / 3, distance / 3));
+        List<Entity> apiEntities = new ArrayList<>();
+        for (net.minecraft.entity.Entity entity : nmsEntities) {
+            apiEntities.add((Entity) entity);
+        }
+        return ImmutableList.copyOf(apiEntities);
```

Comment:
Since Minecraft's `Entity` implements the interface you can directly do `ImmutableList.copyOf` on the `nmsEntities` for some performance boosts